simde/x86/sse4.2.h

```diff
@@ -241,6 +241,57 @@ simde_mm_cmpistrz_16_(simde__m128i b) {
   #define _mm_cmpistrz(a, b, imm8) simde_mm_cmpistrz(a, b, imm8)
 #endif
 
+SIMDE_FUNCTION_ATTRIBUTES
+int64_t bitreflection(int64_t n) {
+  int64_t rev = 0;
+  while(n > 0){
+    rev <<= 1;
+    if(n&1)
+      rev^=1;
+    n >>= 1;
+  }
+  return rev;
+}
+
+SIMDE_FUNCTION_ATTRIBUTES
+int64_t mod2div(int64_t dividend, int64_t divisor){
+  int64_t remainder = dividend;
+  int64_t sdata = floor(log2(dividend))+1;
+  int64_t sdiv = floor(log2(divisor))+1;
+  divisor<<=(sdata-sdiv);
+  int64_t leftdigit = 1LL<< (sdata-1);
+  for(int bit = 0; bit < sdata ; bit++){
+    if(remainder & leftdigit)
+    {
+      remainder^=divisor;
+    }
+    remainder=(remainder<<1);
+  }
+  return (remainder >> (sdata-sdiv+1));
+}
+
+SIMDE_FUNCTION_ATTRIBUTES
+unsigned int
+simde_mm_crc32_u8(unsigned int crc, unsigned char v) {
+#if defined(SIMDE_X86_SSE4_2_NATIVE)
+  return _mm_crc32_u8(crc, v);
+#else
+int64_t tmp1, tmp2, tmp3, tmp4, tmp5, tmp6, dst;
```

Comment:
You really don't need all these temporary variables.  Some operations can be combined (e.g., `tmp1 = bitreflection(v) << 32;`), but even if you don't do that you can reuse variables (`tmp1 = bitreflection(v); tmp1 <<= 32;`).

I'd probably do something like:

```c
int64_t tmp = (bitreflection(v) << 32)  ^ (bitreflection(crc) << 8)
tmp = mod2div(tmp, INT64_C(0x11EDC6F41));
return HEDLEY_STATIC_CAST(int32_t, bitreflection(tmp));
```